Sources/web3swift/HookedFunctions/Web3+BrowserFunctions.swift

```diff
@@ -169,26 +169,28 @@ extension web3.BrowserFunctions {
     public func signTransaction(_ trans: EthereumTransaction, transactionOptions: TransactionOptions, password: String = "web3swift") -> String? {
         do {
             var transaction = trans
-            guard let from = transactionOptions.from else {return nil}
-            guard let keystoreManager = self.web3.provider.attachedKeystoreManager else {return nil}
-            guard let gasPricePolicy = transactionOptions.gasPrice else {return nil}
-            guard let gasLimitPolicy = transactionOptions.gasLimit else {return nil}
-            guard let noncePolicy = transactionOptions.nonce else {return nil}
+            var gasOptions = TransactionOptions()
+            guard let from = transactionOptions.from else { return nil }
+            guard let keystoreManager = self.web3.provider.attachedKeystoreManager else { return nil }
+            guard let gasPricePolicy = transactionOptions.gasPrice else { return nil }
+            guard let gasLimitPolicy = transactionOptions.gasLimit else { return nil }
+            guard let noncePolicy = transactionOptions.nonce else { return nil }
```

Comment:
Wondering if we should treat the absence of nonce policy and other parameters differently.
By that I mean using default values like in the function defined above. That is exactly what's happening there.
```
...
transactionOptions.value = options.value ?? 0
transactionOptions.gasLimit = options.gasLimit ?? .automatic
transactionOptions.gasPrice = options.gasPrice ?? .automatic
```

Or if leaving things as they are now at least we should add swift doc stating that `gasPrice`, `gasLimit`, `nonce` and so on must not be nil.

Reply:
It actually came that way from develop, Here I just added some spacing around the `return nil`. Having said that I can certainly change that, as I think you are correct. Though at a higher level I think we need to revisit how this is handled as once you at EIP-1559 into the mix `gasPrice` has no meaning. As it is the status quo I'm going to leave it, it has been that way since at least 2019, for the time being. But certainly needs to be on the radar once any sort of automatic gas pricing strategy is added into the mix. [automatic gas is still on the todo for this PR]